uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/apis/leave-request-api.js

```diff
@@ -183,6 +183,33 @@ define([
         },
 
         /**
+         * Get leave request for the given id
+         *
+         * @param {object} id - leave request id
+         *
+         * @return {Promise} resolves with {Object}
+         */
+        find: function (id) {
+          $log.debug('LeaveRequestAPI.find');
+          var deferred = $q.defer();
```

Comment:
We need to reject the promise so that the chaining is done properly. Just to make sure I tried with this code, and it threw an error in the console and the chain broke.